datanode/gateway/graphql/proposal_resolver.go

```diff
@@ -88,11 +89,20 @@ func (r *proposalResolver) Terms(_ context.Context, data *types.GovernanceData)
 	return data.Proposal.Terms, nil
 }
 
-func (r *proposalResolver) Votes(_ context.Context, obj *types.GovernanceData) (*ProposalVotes, error) {
-	if obj == nil {
+func (r *proposalResolver) Votes(ctx context.Context, obj *types.GovernanceData) (*ProposalVotes, error) {
+	if obj == nil || obj.Proposal == nil {
```

Comment:
I don't think this is right as it only solves it for the graphQL endpoint and not the call rpc call `ListGoveranceData`. We should already have the vote data in `obj` by now.

If you look at the call to `func (t *tradingDataServiceV2) GetGovernanceData()` we end up calling `t.governanceService.GetNoVotesForProposal(ctx, proposal.ID.String())` to populate the votes.

I believe we just need a similar call in the `ListGoveranceData()` probably in `func (p Proposal) ToProtoEdge()`.

Reply:
@wwestgarth Yeah, that was my initial idea, to just do the same thing as with `GetGovernanceData()`, however the problem I had with that is that the query can end up being a bit "expensive". For every single proposal we would be querying a list of votes, even worse separate query for "yes" and for "no" votes.

So I thought maybe just pull the votes when GQL asks for them. Another way would be to pass a flag down to rpc to ask to include the votes (or not), but that seemed hacky and not consistent with the rest of the endpoints.

Reply:
Yea I don't really know the perf impact (although I suspect the total proposal/vote data will never grow that large really) -- I'll delegate to the postgres-optimisation-pro @pscott31. Any ideas why we do a separate query to get the votes, and not just one query that looks at two tables?

Reply:
@pscott31 Any ideas on this?

Reply:
I think we should do it in the GRPC endpoint like we do in `GetGovernanceData` (I think I suggested this on slack IIRC). If it turns out to be too expensive we can push it down and do it all in one query, but I think there are sufficiently few of these that it doesn't really matter